applicationinsights.ts

```diff
@@ -53,16 +53,16 @@ export let defaultClient: TelemetryClient;
  * and start the SDK.
  */
 export function setup(instrumentationKey?: string) {
-    if(!defaultClient) {
-        defaultClient = new TelemetryClient(instrumentationKey);
-        _console = new AutoCollectConsole(defaultClient);
-        _exceptions = new AutoCollectExceptions(defaultClient);
-        _performance = new AutoCollectPerformance(defaultClient);
-        _serverRequests = new AutoCollectHttpRequests(defaultClient);
-        _clientRequests = new AutoCollectHttpDependencies(defaultClient);
-    } else {
-        Logging.info("The default client is already setup");
+    if(defaultClient) {
+        Logging.info("setup was called previously. Overwriting existing defaultClient");
+        dispose();
     }
+    defaultClient = new TelemetryClient(instrumentationKey);
```

Comment:
Even though you call dispose, I'd still be concerned about a reference to the old client sticking around somewhere - preventing gc. Consider resetting the config on the existing telemetryclient rather than constructing from scratch.

Reply:
Updated to only reconfigure. All of the other objects are okay to reconstruct since they shouldn't have references elsewhere